src/renderer/entities/network/model/network-model.ts

```diff
@@ -174,77 +264,266 @@ sample({
+  source: {
+    apis: $apis,
+    subscriptions: $metadataSubscriptions,
+  },
+  fn: ({ apis, subscriptions }, statuses) =>
+    Object.entries(apis).filter(([chainId]) => {
+      return statuses[chainId as ChainId] === ConnectionStatus.CONNECTED && !subscriptions[chainId as ChainId];
+    }) as [ChainId, ApiPromise][],
+  target: subscribeMetadataFx,
 });
 
 sample({
-  clock: disconnectStarted,
-  source: { providers: $providers, apis: $apis },
-  filter: ({ providers, apis }, chainId) => Boolean(providers[chainId] && apis[chainId]),
-  fn: ({ providers, apis }, chainId) => ({
-    provider: providers[chainId],
-    api: apis[chainId],
-    chainId: chainId,
+  clock: $connectionStatuses,
+  source: {
+    subscriptions: $metadataSubscriptions,
+  },
+  fn: ({ subscriptions }, statuses) =>
+    Object.entries(subscriptions).filter(([chainId]) => {
+      return statuses[chainId as ChainId] === ConnectionStatus.DISCONNECTED && subscriptions[chainId as ChainId];
+    }) as [ChainId, UnsubscribePromise][],
+  target: [unsubscribeMetadataFx, metadataUnsubscribed],
+});
+
+sample({
+  clock: metadataUnsubscribed,
+  source: $metadataSubscriptions,
+  fn: (subscriptions, unsubscribes) => {
+    unsubscribes.forEach(([chainId]) => {
+      delete subscriptions[chainId as ChainId];
+    });
+
+    return subscriptions;
+  },
+  target: $metadataSubscriptions,
+});
+
+sample({
+  clock: subscribeMetadataFx.doneData,
+  source: $metadataSubscriptions,
+  fn: (subscriptions, unsubscribes) => {
+    unsubscribes.forEach(([chainId, cb]) => {
+      subscriptions[chainId as ChainId] = cb;
+    });
+
+    return subscriptions;
+  },
+  target: $metadataSubscriptions,
+});
+
+// =====================================================
+// =============== Connection section ==================
+// =====================================================
+
+$connections
+  .on(populateConnectionsFx.doneData, (_, connections) => {
+    return keyBy(connections, 'chainId');
+  })
+  .on(createConnectionFx.doneData, (state, connection) => {
+    return connection
+      ? {
+          ...state,
+          [connection.chainId]: connection,
+        }
+      : state;
+  })
+  .on(deleteConnectionFx.doneData, (state, connectionId) => {
+    const deletedConnection = Object.values(state).find((c) => c.id === connectionId);
+
+    if (!deletedConnection?.chainId) return state;
+
+    const { [deletedConnection.chainId]: _, ...newConnections } = state;
+
+    return newConnections;
+  })
+  .on(updateConnectionFx.doneData, (state, connection) => {
+    return {
+      ...state,
+      [connection.chainId]: connection,
+    };
+  });
+
+forward({
+  from: kernelModel.events.appStarted,
+  to: populateConnectionsFx,
+});
+
+sample({
+  clock: rpcNodeAdded,
+  source: $connections,
+  fn: (connections, { chainId, rpcNode }) => {
+    const connection = connections[chainId];
+
+    connection.customNodes = (connection.customNodes || []).concat(rpcNode);
```

Comment:
Not sure that mutating `connection` is a good idea, we might produce bugs this way

Some samples produce new object without inner mutations, for example this one  -`lightClientSelected`